x-pack/plugins/canvas/common/lib/escape_reg_exp.js

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function escapeRegExp(string) {
```

Comment:
Any harm in `TS`ing this?

Reply:
This file is really magical... I'd appreciate some inline docs about what this is used for, and under what circumstances.